src/2d/spreadinterp2d.cu

```diff
@@ -565,13 +570,13 @@ void Interp_2d_NUptsdriven(FLT *x, FLT *y, CUCPX *c, CUCPX *fw, int M, const int
 		cnow.y = 0.0;
 		for(int yy=ystart; yy<=yend; yy++){
 			FLT disy=abs(y_rescaled-yy);
-			FLT kervalue2 = evaluate_kernel(disy, es_c, es_beta);
+			FLT kervalue2 = disy<ns/2.0 ? evaluate_kernel(disy, es_c, es_beta):0;
```

Comment:
Hi, while this works, I suggest you incorporate this check into evaluate_kernel. This requires adding an extra arg, ns. But then you won't have the same check repeated 10 times as below, which is much harder to maintain...